woocommerce/google-listings-and-ads
# Connect programs report table to API data

### Changes proposed in this Pull Request:

- Separate category-agnostic `CompareTableCard` (eda8cc4, 01df46e, bf32393)
- Use it in `CompareProductsTableCard` (bf32393)
- Use it in `CompareProgramsTableCard` and connect it to API data (0dde6b3)

Other changes:
- Unify `trackEventId` property name between programs and products report pages (2af2f06)
- Unify `isLoading` parametter across all `<*TableCard>`s. (bf32393)
- Add support for quering programs report by ids. (7e7af54)

Implements part of https://github.com/woocommerce/google-listings-and-ads/issues/242

### Screenshots:

![Programs report table](https://user-images.githubusercontent.com/17435/118701853-7a8f4b00-b814-11eb-96b6-a4b8266348f9.gif)



### Detailed test instructions:
0. Make sure you have some test data, or use the mock:
```php
add_filter(
	'gla_prepared_response_ads-reports-products',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/ads/reports/products.json' ), true ) ?: [];
	},
);
add_filter(
	'gla_prepared_response_mc-reports-products',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/mc/reports/products.json' ), true ) ?: [];
	},
);
add_filter(
	'gla_prepared_response_ads-reports-programs',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/ads/reports/programs.json' ), true ) ?: [];
	},
);
add_filter(
	'gla_prepared_response_mc-reports-programs',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/mc/reports/programs.json' ), true ) ?: [];
	},
);
```
1. Go to https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fprograms
2. Check the data in the table
3. Filter programs "All Google Programs", "Single: Free", "Single: Campa", "Compare"
4. Check that unavailable columns collapse when you query only for free listings.
4. Check that "Unavailable" is shown, if programs are missing some data if you query for many programs.
5. Check that once you select programs to compare respective `ids` are being sent in `/wp-json/wc/gla/ads/reports/programs` requests

:warning:

1.  Make sure the Product table works as before.

### Changelog Note:

> Connected programs report table to API data.

### :scroll: To be done here, or in a separate issue:
1. Make report hooks adapt the request and returned data when the user selects to order by a field that's not available for a free listing.
